crates/client/rpc-core/src/utils.rs

```diff
@@ -89,15 +89,16 @@ fn to_legacy_entry_point(entry_point: EntryPoint) -> Result<LegacyContractEntryP
 }
 
 /// Returns the current Starknet block from the block header's digest
-pub fn get_block_by_block_hash<B, C>(client: &C, block_hash: <B as BlockT>::Hash) -> Option<StarknetBlock>
+pub fn get_block_by_block_hash<B, C>(client: &C, block_hash: <B as BlockT>::Hash) -> Result<StarknetBlock>
```

Comment:
let's throw a `StarknetRpcApiError`?

Reply:
@apoorvsadana `StarknetRpcApiError` is defined in the `mc-rpc` crate, `get_block_by_block_hash` is defined in `mc-rpc-core`, and currently `mc-rpc-core` is a dependency of `mc-rpc`. It will cause a circular dependency between two crates if  throwing a `StarknetRpcApiError`. Do you have any suggestions? Thanks

Reply:
Understood, we can keep it as it is. Can we log the error in this function before returning it?